src/Query/JoinClause.php

```diff
@@ -116,15 +125,36 @@ public function using(...$columns): self
     }
 
     /**
-     * Alias for using method.
+     * Set "on" clause for join.
      *
-     * @param array ...$columns
+     * @param Closure|string $first
+     * @param mixed|null     $operator
+     * @param mixed|null     $second
+     * @param string         $boolean
      *
      * @return JoinClause
      */
-    public function on(...$columns): self
+    public function on($first, $operator = null, $second = null, string $concatOperator = Operator::AND): self
```

Comment:
Why it is possible to pass something other than a string and why there is only one required parameter? Does clickhouse support joins without comparison of two columns?

I mean, for example this kind of joins will work?
```sql
select * from table1 left any join column;
```

Reply:
Query builder shouldn't give the possibility to write invalid queries.